### PR TITLE
PlayStation Portable support

### PR DESCRIPTION
**Pull request by AJenbo**
Just enough hacks to get it building. Needs a bit of cleaning, packing testing, and proper implementations of stubs.

Depends on https://github.com/diasurgical/devilutionX/pull/1314

Thanks to @glebm and @dallemon

### CODE REVIEW

**Comment by sajattack** [comment]
No shoutouts to your buddy @sajattack ? :sweat_smile:

File: SourceX/DiabloUI/ttf_render_wrapped.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ SDL_Surface *RenderUTF8_Solid_Wrapped(TTF_Font *font, const char *text, SDL_Colo
 			return NULL;
 		}
 
-#ifdef __3DS__
+#if 1
```
Comment (sajattack) [inline SourceX/DiabloUI/ttf_render_wrapped.cpp:60]: btw you can use `#ifdef PSP` here I think

**Comment by AJenbo** [comment]
@sajattack sorry I though you had enough of this port for one day :sweat_smile:

File: CMakeLists.txt [hunk 2]
```diff
@@ -531,6 +531,7 @@ endforeach(def_name)
 genex_for_option(ASAN)
 genex_for_option(UBSAN)
 foreach(target devilution ${BIN_TARGET} SourceS)
+target_link_libraries(${target} PRIVATE stdc++ pspgum SDL2_ttf freetype SDL2_image png jpeg SDL2_mixer vorbisfile vorbis ogg psphprm mikmod pspaudiocodec pspaudio SDL2main SDL2 g c GL GLU glut z m pspvfpu pspvram  -L${PSPDEV}/psp/sdk/lib -L${PSPDEV}/psp/lib pspdebug pspgu pspctrl pspge pspirkeyb psppower psppower_driver pspdisplay psphprm pspsdk psprtc psputility pspnet_inet  c pspuser)
```
Comment (glebm) [inline CMakeLists.txt:534]: linker order matters
e.g. you have to link `c` last

move this where the `if(SWITCH)` is

Comment (AJenbo) [reply CMakeLists.txt:534]: most of the work was spent getting this in an order that would even link :D

Comment (galaxyhaxz) [reply CMakeLists.txt:534]: At least, using the PSP sdk in proper, you'd have a makefile which would make it a bit more readable then having to cram it all on one line. You usually start with static libraries, then add the stubs for the kernel imports and/or sdk modules.
```
LDLIBS		= -lgum -lgu -lwave -lm
LOADLIBES	=  $(LIBDIR)/ctrl_stub.a
LOADLIBES	+= $(LIBDIR)/display_stub.a
LOADLIBES	+= $(LIBDIR)/ge_user_stub.a

TARGET	= balloon
OBJS	= main.o balloon.o cloud.o sphere.o pole.o

all:	$(TARGET).elf
```

**Comment by AJenbo** [comment]
Replaced by https://github.com/diasurgical/devilutionX/pull/5869

**Pull request closed without merging**